asv/www/graphdisplay.js

```diff
@@ -1017,28 +1047,43 @@ $(document).ready(function() {
             return;
 
         if (x_coordinate_axis == 0) {
+            if (!date_scale) {
+                // restore original graphs
+                graphs = orig_graphs;
+                options.xaxis.mode = null;
+                options.xaxis.axisLabel = 'commits';
+                options.xaxis.tickFormatter = function (v, axis) { return ""; };
+            }
             if (even_spacing) {
-                even_spacing_dates = [];
+                even_spacing_revisions = [];
                 $.each(graphs, function(i, graph) {
                     var data = graph.data;
                     for (var j = 0; j < data.length; ++j) {
-                        if (even_spacing_dates.indexOf(data[j][0]) === -1) {
-                            even_spacing_dates.push(data[j][0]);
+                        if (even_spacing_revisions.indexOf(data[j][0]) === -1) {
+                            even_spacing_revisions.push(data[j][0]);
                         }
                     }
                 });
-                even_spacing_dates.sort(function(a, b) { return a - b; });
-
-                options.xaxis.axisLabel = 'commits';
+                even_spacing_revisions.sort(function(a, b) { return a - b; });
                 options.xaxis.transform = even_spacing_transform;
                 /* inverseTransform is required for plothover to work */
                 options.xaxis.inverseTransform = even_spacing_inverseTransform;
-                options.xaxis.tickFormatter = function (v, axis) {
-                    return "";
-                };
-            } else {
+            } else if (date_scale) {
                 options.xaxis.mode = 'time';
                 options.xaxis.axisLabel = 'commit date';
+                options.xaxis.tickFormatter = null;
+
+                // deep copy graphs and transform into time series data
+                graphs = $.extend(true, [], orig_graphs);
+                date_to_revision = {};
+                $.each(graphs, function(i, graph) {
+                    $.each(graph.data, function(j, point) {
+                        var date = $.asv.master_json.revision_to_date[point[0]];
+                        date_to_revision[date] = point[0];
+                        point[0] = date;
+                    });
+                    graph.data.sort(function(a, b) { return a[0] - b[0]; });
+                });
```

Comment:
Some hints to understand this `orig_graphs` purpose.

I was planning to use flot `transform` and `inverseTransform` functions build the `date scale` graph using `revision_to_date`. But flot assume the transform function does not reorder values (eg. is monotone), so it cannot be used here unfortunately.

So this build a new graphs and keep a reference to the old graphs for switching from one to another (but maybe you have a better idea ?).